src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -274,6 +274,101 @@ function svd_rev(USV::SVD, Ū, s̄, V̄)
     return Ā
 end
 
+#####
+##### `qr`
+#####
+
+
+function ChainRules.rrule(::typeof(getproperty), F::LinearAlgebra.QRCompactWY, d::Symbol)
+    function getproperty_qr_pullback(Ȳ)
+        # The QR factorization is calculated from `factors` and T, matrices stored in the QRCompactWYQ format, see
+        # R. Schreiber and C. van Loan, Sci. Stat. Comput. 10, 53-57 (1989).
+        # Instead of backpropagating Q̄ and R̄ through (factors)bar and T̄, we re-use factors to carry Q̄ and T to carry R̄
+        # in the Tangent object.
+        ∂T = d === :R ? Ȳ : nothing
```

Comment:
We do not uses `nothing` to represent not used.
We use `ZeroTangent` for not used, (and `NoTangent` for not having a meaningful tangent space)